### ISSUE TITLE
Duplicate <instance> elements when using multiple external instances

### ISSUE THREAD

**Issue by sjudeng** [XLSForm/pyxform#88]
When using `xls2xform` to convert a form containing multiple external instances, the generated XML appears to contain extra `<instance>` elements for the external instances. This causes issues loading/viewing the form in Enketo. If the extra `<instance>` elements are removed the form works fine.

The XLSForm is  [attached](https://github.com/XLSForm/pyxform/files/488184/select_from_file_test2.xlsx) and shown below.

survey:

| type | name | label | choice_filter |
| --- | --- | --- | --- |
| select_one_from_file states.csv | State | Select State |  |
| select_one_from_file cities.csv | City | Select City | state=/select_from_file_test/State |

choices:

| list_name | name | label |
| --- | --- | --- |
| states | name | label |
| cities | name | label |

settings:

| form_title | form_id |
| --- | --- |
| select_from_file test2 | select_from_file_test2 |

Below is the XML generated from `xls2xform`. Note this is with `--skip_validate` because of #87.

``` xml
<?xml version="1.0"?>
<h:html xmlns="http://www.w3.org/2002/xforms" xmlns:ev="http://www.w3.org/2001/xml-events" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:jr="http://openrosa.org/javarosa" xmlns:orx="http://openrosa.org/xforms" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <h:head>
    <h:title>select from file test2</h:title>
    <model>
      <itext>
        <translation default="true()" lang="default">
          <text id="static_instance-cities-0">
            <value>label</value>
          </text>
          <text id="static_instance-states-0">
            <value>label</value>
          </text>
        </translation>
      </itext>
      <instance>
        <select_from_file_test2 id="select_from_file_test2">
          <State/>
          <City/>
          <meta>
            <instanceID/>
          </meta>
        </select_from_file_test2>
      </instance>
      <instance id="states">
        <root>
          <item>
            <itextId>static_instance-states-0</itextId>
            <name>name</name>
          </item>
        </root>
      </instance>
      <instance id="cities">
        <root>
          <item>
            <itextId>static_instance-cities-0</itextId>
            <name>name</name>
          </item>
        </root>
      </instance>
      <instance id="states" src="jr://file-csv/states.csv">
        <root>
          <item>
            <name/>
            <label/>
          </item>
        </root>
      </instance>
      <instance id="cities" src="jr://file-csv/cities.csv">
        <root>
          <item>
            <name/>
            <label/>
          </item>
        </root>
      </instance>
      <bind nodeset="/select_from_file_test2/State" type="select1"/>
      <bind nodeset="/select_from_file_test2/City" type="select1"/>
      <bind calculate="concat('uuid:', uuid())" nodeset="/select_from_file_test2/meta/instanceID" readonly="true()" type="string"/>
    </model>
  </h:head>
  <h:body>
    <select1 ref="/select_from_file_test2/State">
      <label>Select State</label>
      <itemset nodeset="instance('states')/root/item">
        <value ref="name"/>
        <label ref="label"/>
      </itemset>
    </select1>
    <select1 ref="/select_from_file_test2/City">
      <label>Select City</label>
      <itemset nodeset="instance('cities')/root/item[state=/select_from_file_test2/State]">
        <value ref="name"/>
        <label ref="label"/>
      </itemset>
    </select1>
  </h:body>
</h:html>
```

Note there are duplicate `<instance>` elements defined for each of the external instances (e.g. states and cities). I think the first ones (see below) should not be present in the XML. If I remove these instance elements then the form loads and works as expected in Enketo.

``` xml
      <instance id="states">
        <root>
          <item>
            <itextId>static_instance-states-0</itextId>
            <name>name</name>
          </item>
        </root>
      </instance>
      <instance id="cities">
        <root>
          <item>
            <itextId>static_instance-cities-0</itextId>
            <name>name</name>
          </item>
        </root>
      </instance>
```

**Comment by MartijnR** [comment]
Thanks for this! 

There are duplicate references to 'states' and 'cities' in the XLSForm (choices sheet, and select_one_from_file). The choices sheet lists should be ignored in this case by pyxform, so it's a bug.

**Comment by lindsay-stevens** [comment]
I think this was fixed in #169, commit f01917f. If not, please re-open.
